src/main/java/com/epam/eco/commons/kafka/consumer/bootstrap/BootstrapConsumerThread.java

```diff
@@ -52,8 +59,15 @@ public void run() {
                 handler.accept(consumer.fetch());
                 bootstrapLatch.countDown();
             }
+        } catch (Exception e) {
+            BootstrapException wrapped = toBootstrapException(e);
+            log.error(wrapped.getMessage(), wrapped);
+            exception.set(wrapped);
+            // Trigger latch to rethrow exception immediately
+            bootstrapLatch.countDown();
         } finally {
             running.set(false);
+            consumer.close();
```

Comment:
could an error be thrown here? maybe we need to wrap this call to some kind of 'doQuietly'?

Reply:
bootstrap consumer already wraps close call with `doQuietly`